sympy/polys/ring_series.py

```diff
@@ -572,6 +574,111 @@ def _coefficient_t(p, t):
             p1[monomial_div(expv, expv1)] = p[expv]
     return p1
 
+def rs_fast_series_reversion(p,x,n):
+
+    r"""
+    Fast series reversion (aka Lagrange Inversion) using Newton-like updates.
+
+    Given a ring polynomial $p$ in variable $x$, this method computes a ring polynomial $q$ in $x$ such that
+    $q(p(x)) = x + O(x^n)$.
+
+    The series should be of form $p(x) = a_1 x + a_2 x^2 + ...$ where $a_1$ is invertible.
+
+    Parameters
+    ==========
+
+    p : :class:`~.PolyElement` to be inverted.
+    x : :class:`~.PolyElement` with respect to which ``p`` is inverted.
+    n : number of expansion terms to be computed.
+
+    Note
+    ====
+
+    The algorithm adapts the Newton method to solve a general equation
+    $$
+    f(x)=y.
+    $$
+    By Taylor's expansion:
+    $$
+    f(x+h) = f(x) + f'(x) h + O(h^2).
+    $$
+    If $f(x) = y + O(y^{k+1})$ then with $h = -\frac{f(x)-y}{f'(x)} = O(y^{k+1})$ we obtain:
+    $$
+    f(x+h)-y = O(h^2) = O(y^{2k+2}).
+    $$
+    Thus, updating $x\gets x-h$ increases the accuracy from $k$ to $2k+1$ terms.
+
+    Examples
+    ========
+
+    The example below shows how to invert the expresion $f=\mathrm{e}^{x}-1$ around $x=0$.
+    First we truncate the series of $f$, then convert it to fit the polynomial ring - in this case $Q[x]$, and finally call the method:
+
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+    >>> from sympy.polys.ring_series import rs_fast_series_reversion
+    >>> f = exp(x)-1
+    >>> n = 4
+    >>> f = f.series(x,0,n).removeO()
+    >>> Rx = f.as_poly(x,field=True).domain[x]
+    >>> fr = Rx.from_sympy(f)
+    >>> xr = Rx.from_sympy(x)
+    >>> rs_fast_series_reversion(fr,xr,n)
+    1/3*x**3 - 1/2*x**2 + x
+
+    The example below show few first explicit coefficients of the inversion expansion, namely:
+
+    >>> import sympy as sm
+    >>> from sympy.polys.ring_series import rs_fast_series_reversion
+    >>> a = sm.IndexedBase('a')
+    >>> x = sm.symbols('x')
+    >>> f = sum(a[k]*x**k for k in range(1,4))
+    >>> R = f.as_poly(x,field=True).domain
+    >>> Rx = R[x]
+    >>> fr = Rx.from_sympy(f)
+    >>> xr = Rx.from_sympy(x)
+    >>> rs_fast_series_reversion(fr,xr,n).as_expr()
+    x**3*(-a[3]/a[1]**4 + 2*a[2]**2/a[1]**5) - x**2*a[2]/a[1]**3 + x/a[1]
+
+    References
+    ==========
+
+    .. [1] Brent, Richard P., and Hsiang T. Kung. „Fast algorithms for manipulating formal power series.” Journal of the ACM (JACM) 25, no. 4 (1978): 581-595.
+    .. [2] Weisstein, Eric W. "Series Reversion." From MathWorld--A Wolfram Web Resource. https://mathworld.wolfram.com/SeriesReversion.html
+    """
+
+    f = p
+    Rx = f.ring
+    df = f.diff(x)
+    df_inv = rs_series_inversion(df,x,n)
+
+    # inject the auxiliary variable y
+    y = Dummy('y')
+    Rx = Rx.to_domain()
+    Rxy = Rx.inject(y)
+    df_inv = Rxy.convert(df_inv)
+    xr = Rxy.convert(x)
+    f = Rxy.convert(f)
+    yr = Rxy.from_sympy(y)
```

Comment:
When calling `convert` it is better to specify the domain if it is known e.g. `Rxy.convert_from(df_inv, Rx)`.